IBAnimatable/Chainable.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  Chainable.swift
+//  IBAnimatable
+//
+//  Created by jason akakpo on 01/01/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+typealias AnimationTuple = (type: AnimationType, conf: AnimationConfiguration)
+
+struct AnimationConfiguration {
+  let damping: CGFloat
+  let velocity: CGFloat
+  let duration: Double
+  let delay: Double
+  let force: CGFloat
+
+}
+
+
+public class AnimationPromise<T: UIView where T: Animatable> {
+
+  private var view: T
+  private var animationList: [AnimationTuple] = []
```

Comment:
Don't know what do you think about that syntax, but I personally prefer: `private var animationList = [AnimationTuple]()` since it's more concise 
Any opinions on this @JakeLin ?